src/content/developers/tutorials/submitting-and-verifying-your-smart-contract-to-etherscan/index.md

```diff
@@ -0,0 +1,263 @@
+
+We need the `truffle-plugin-verify` plugin to automatically verify your truffle smart contract's source code and ABI on Etherscan. In your project directory run:
+
+```text
+npm install -g truffle-plugin-verify
+```
+
+Once installed, add the plugin to your `truffle-config.js` file. Your file should look similar to this. 
+
+```javascript
+const HDWalletProvider = require("@truffle/hdwallet-provider");
+require('dotenv').config()
+
+const { API_URL, MNEMONIC } = process.env;
+
+module.exports = {
+  networks: {
+    development: {
+      host: "localhost",
+      port: 8545,
+      network_id: "*", // Match any network id
+      gas: 5000000
+    },
+    ropsten: {
+      provider: function() {
+        return new HDWalletProvider(MNEMONIC, API_URL)
+      },
+      network_id: 3
+    }
+  },
+  compilers: {
+    solc: {
+      settings: {
+        optimizer: {
+          enabled: true, // Default: false
+          runs: 200      // Default: 200
+        },
+      }
+    }
+  },
+  plugins: ['truffle-plugin-verify'] //PLUGIN ADDED HERE
+};
+```
+
+You're almost at the finish line! 😅
+
+Let's update your `truffle-config.js` file to include your Etherscan API key. See the bottom of the code below for reference:
+
+```javascript
+const HDWalletProvider = require("@truffle/hdwallet-provider");
+require('dotenv').config()
+
+const { API_URL, MNEMONIC } = process.env;
+
+module.exports = {
+  networks: {
+    development: {
+      host: "localhost",
+      port: 8545,
+      network_id: "*", // Match any network id
+      gas: 5000000
+    },
+    ropsten: {
+      provider: function() {
+        return new HDWalletProvider(MNEMONIC, API_URL)
+      },
+      network_id: 3
+    }
+  },
+  compilers: {
+    solc: {
+      settings: {
+        optimizer: {
+          enabled: true, // Default: false
+          runs: 200      // Default: 200
+        },
+      }
+    }
+  },
+  plugins: ['truffle-plugin-verify'],
+  api_keys: {
+    etherscan: process.env.ETHERSCAN_API_KEY
+  }
+
+};
+```
+
+#### Step 3.2 Verify your smart contract on Etherscan! {#step-3.2-verify-your-smart-contract-on-etherscan}
+
+Last but not least, run the following command in your terminal:
+
+```text
+truffle run verify HelloWorld --network ropsten
+```
+
+If all goes well, you should see the following message in your terminal.
+
+```text
+Verifying HelloWorld 
+Pass - Verified: https://ropsten.etherscan.io/address/<contract-address>#contracts
+Successfully verified 1 contract(s).
+```
+
+### Step 4: Check out your smart contract on Etherscan! {#step-4-check-out-your-smart-contract-on-etherscan!}
+
+When you navigate to the link provided in your terminal, you should be able to see your smart contract code and ABI published on Etherscan!
+
+![](https://gblobscdn.gitbook.com/assets%2F-MB17w56kk7ZnRMWdqOL%2F-MReibSVpLlOCa_Si494%2F-MRewuf9NzToMPg5WIln%2Fimage.png?alt=media&token=ca99b37e-6847-4b7a-8fef-371783f2e536)
```

Comment:
We should host any images we include on ethereum.org so we're not relying on third parties for their availability ([Check out the style guide for an example](https://ethereum.org/en/contributing/style-guide/#images)). Could you upload this image in the local directory and reference it that way instead?

```suggestion
![](https://gblobscdn.gitbook.com/assets%2F-MB17w56kk7ZnRMWdqOL%2F-MReibSVpLlOCa_Si494%2F-MRewuf9NzToMPg5WIln%2Fimage.png?alt=media&token=ca99b37e-6847-4b7a-8fef-371783f2e536)
```